src/hooks/resources/device.ts

```diff
@@ -365,15 +365,19 @@ sbvrUtils.addPureHook('PATCH', 'resin', 'device', {
 			args.request.values.device_name != null
 		) {
 			waitPromises.push(
-				affectedIds.then(deviceIds =>
-					postDevices({
+				affectedIds.then(deviceIds => {
+					if (deviceIds.length === 0) {
```

Comment:
Had to also add this check because with the unnecessary write avoidance suddenly no devices get affected :P